### PR TITLE
build: Update DEB and RPM artifacts to align with documentation/non contrib collector

### PR DESCRIPTION
**Pull request by Jammicus**
**Description:** Updating RPM and DEB artifacts to match documentation

Currently the RPM and Deb artifacts are not aligned to how they are described in the [documentation](https://opentelemetry.io/docs/collector/getting-started/#linux-packaging). They are:

- Not bundled with a default configuration file, so the service automatically fails on start when ran via systemd. https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/3097
- Not bundled with a environment file for the systemd service, so users cannot follow the documentation steps for adding additional options to the service (https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/3143)

This PR does the following:

- Bundles the default configuration file in both the RPM and DEB artifacts as 
- Bundles a default environment file with `OTELCOL_OPTIONS` 
- Removes now redundant test that checks whether the service will crash without a config file. As its bundled with the artifacts it should never be without a config file unless the users remove it. (Happy to add back if you feel its needed)

**Link to tracking Issue:** 
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/3097
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/3143

**Testing:**  Automated tests have installed + checked service is running after 5 seconds. Manually tested the DEB on Ubuntu 18.04

**Documentation:** N/A

Edit: Updating path. 
Edit 2: Including https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/3143 as its dependent on the config being present

### CODE REVIEW

**Comment by pranavmarla** [comment]
@Jammicus Thanks for looking into this! 

Quick question: Since this is the _contrib_ collector, should the config be located at:
`/etc/otel-collector/config.yaml`
OR
`/etc/otel-contrib-collector/config.yaml`?

**Comment by Jammicus** [comment]
> @Jammicus Thanks for looking into this!
> 
> Quick question: Since this is the _contrib_ collector, should the config be located at:
> `/etc/otel-collector/config.yaml`
> OR
> `/etc/otel-contrib-collector/config.yaml`?


> @Jammicus Thanks for looking into this!
> 
> Quick question: Since this is the _contrib_ collector, should the config be located at:
> `/etc/otel-collector/config.yaml`
> OR
> `/etc/otel-contrib-collector/config.yaml`?
> 
> Update: Just confirmed by looking at the agent error message in my issue (#3097) -- when the agent fails due to a missing config file, the error message specifically mentions that it's looking for the config file in `/etc/otel-contrib-collector/config.yaml`. Since it seems to be looking there by default, I assume that's where the default config file should be located.

Oooh good spot! Will update

**Comment by Jammicus** [comment]
Looking at the failing tests, its failing due to a test which explicitly checks that the collector does not start without config:

```
# test install without config, service should not be running
echo
$docker_run
install_pkg $container_name "$PKG_PATH"
sleep 5
echo "Checking $SERVICE_NAME service status after install without config ..."
if $docker_exec systemctl --no-pager status $SERVICE_NAME; then
    echo "$SERVICE_NAME service is running when it should not" >&2
    exit 1
fi
echo "$SERVICE_NAME service is correctly not running"
```

Looking at the file this seems to have been added when it was initially created. Will need some guidance from the contributors as to whether this is intentional and the collector should not be shipped with default config or can be removed (Happy to update/close this depending on your guidance)

File: internal/buildscripts/packaging/fpm/otel-contrib-collector.conf
Comment (Jammicus) [reply internal/buildscripts/packaging/fpm/otel-contrib-collector.conf:5]: Thanks, updated line 3. 

Not sure on the naming conventions - approvers can give us some guidance when they have time to review this

**Comment by bogdandrutu** [comment]
@jchengsfx can you review?

File: internal/buildscripts/packaging/fpm/otel-contrib-collector.conf
Comment (anuraaga) [reply internal/buildscripts/packaging/fpm/otel-contrib-collector.conf:5]: I think there's value in keeping the environment variables symmetrical between contrib and non-contrib, they're mostly just different versions of the same thing so more swappability is nice

**Review by anuraaga — APPROVED** [review]

File: internal/buildscripts/packaging/fpm/otel-contrib-collector.conf
Comment (jchengsfx) [reply internal/buildscripts/packaging/fpm/otel-contrib-collector.conf:5]: IMO, I would use `OTELCONTRIBCOL_OPTIONS` just to be consistent with the `otelcontribcol` binary name.

**Comment by bogdandrutu** [comment]
@jchengsfx please approve if everything looks good

**Review by jchengsfx — APPROVED** [review]

**Pull request merged**
